### DISCUSSION TITLE
Loading Dataset with .jsonl Metadata.

### DISCUSSION THREAD

**Discussion by nighting0le01** [huggingface/datasets#6776] [Q&A]
Hi I have the following directory structure:
|--dataset
| |-- images
| |-- metadata1000.csv
| |-- metadata1000.jsonl
| |-- padded_images

Example of metadata1000.jsonl file
{"caption": "a drawing depicts a full shot of a black t-shirt with a triangular pattern on the front there is a white label on the left side of the triangle", "image": "images/212734.png", "gaussian_padded_image": "padded_images/p_212734.png"}
{"caption": "an eye-level full shot of a large elephant and a baby elephant standing in a watering hole on the left side is a small elephant with its head turned to the right  of dry land, trees, and bushes", "image": "images/212735.png", "gaussian_padded_image": "padded_images/p_212735.png"}
.
.
.

I'm trying to use `dataset = load_dataset("imagefolder", data_dir='/dataset/', split='train')` to load the the dataset, however it is not able to load according to the fields in the metadata1000.jsonl .
please assist to load the data properly

**Comment by dexhunter** [comment]
For your existing JSONL schema, load it as JSON, resolve both image paths, and cast both columns to `Image`:

```python
from pathlib import Path
from datasets import Image, load_dataset

root = Path("/dataset")
ds = load_dataset(
    "json",
    data_files=str(root / "metadata1000.jsonl"),
    split="train",
)

def resolve_paths(row):
    return {
        "image": str(root / row["image"]),
        "gaussian_padded_image": str(root / row["gaussian_padded_image"]),
    }

ds = ds.map(resolve_paths)
ds = ds.cast_column("image", Image())
ds = ds.cast_column("gaussian_padded_image", Image())

print(ds[0]["caption"])
image = ds[0]["image"]
padded_image = ds[0]["gaussian_padded_image"]
```

This preserves one row per JSONL record, the caption, and the association between the two images. Paths must point to files available on the machine reading the dataset.

`imagefolder` does not automatically interpret arbitrary JSONL filenames and image-column names. Its [metadata convention](https://huggingface.co/docs/datasets/image_dataset#imagefolder) uses `metadata.jsonl` and fields such as `file_name`/`*_file_name`. Loading your manifest explicitly avoids having to rename or restructure it.
